### DISCUSSION TITLE
Add ability to navigate from related HasOne/MorphOne to Resource Detail page

### DISCUSSION THREAD

**Discussion by oskobri** [laravel/nova-issues#4165] [Ideas & Feature Requests] [answered]
- Laravel Version: 9.11.0
- Nova Version: 4.3.4
- PHP Version: 81.0
- Database Driver & Version: Mysql 8.0

### Description:
With new hasOne/morphOne relationship integrated into the parent view. We cannot access to relationships like hasMany or manyToMany of the hasOne relationship because only replicate/delete and edit resource are available. Is it possible to add the show button ?

![Capture d’écran 2022-05-05 à 09 28 54](https://user-images.githubusercontent.com/43296157/166880518-b8f730c1-3128-467e-9285-d77e14c589e3.png)

**Comment by crynobone** [comment]
Please provide reproducing repository similar to https://github.com/nova-issues

**Comment by oskobri** [comment]
https://github.com/oskobri/nova-hasone

**Comment by oskobri** [comment]
Is it at least possible to go back to the old way when hasOne/morphOne were not built in its parent ?

**Comment by ali-raza-saleem** [comment]
I am facing the same issue with Nova v4.5.4, whereas the issue has been marked resolved in Nova v4.2.3. Can you please check? (https://nova.laravel.com/releases/4.2.3)

**Comment by leigeber** [comment]
Agree with this. Don't mind the new preview layout in most cases, but the ability to click through is critical.

**Comment by ovdahlberg** [comment]
I agree as well. Would be nice to be able to view the resource as before :)

**Comment by leith** [comment]
@crynobone the repo from @oskobri illustrates the issue pretty well. It looks like it would be a relatively small change to resolve:
* copy the view button from `components/ResourceTableRow.vue`
* inject it into `views/Detail.vue` (currently at line 67)
* add a `v-if="viaResource"` condition (or similar) to the copied `<Link>` so that the primary resource doesn't end up with a view button on its own view page

It would probably need some additional tests to make sure that the authorization permutations play nice, too.

**Comment by crynobone** [comment, marked as answer]
This has been implemented in v4.19.0
